source/rrRoadRunner.cpp

```diff
@@ -304,6 +304,40 @@ class RoadRunnerImpl {
         //memset((void*)integrators, 0, sizeof(integrators)/sizeof(char));
     }
 
+    RoadRunnerImpl(const RoadRunnerImpl& rri) :
+        mInstanceID(rri.mInstanceID),
```

Comment:
just to verify: do we need to ensure that there are no duplicate instance ID's, even between cloned instances?

Reply:
Oh actually the instanceID is overwritten in the RoadRunner constructor later, so this wouldn't be a problem.

Reply:
That is an excellent catch!  I just followed what had been done for saveState/loadState, which kept the same mInstanceID when loaded.  But looking at the rest of the code, it seems like the intent is to give each RoadRunner object a unique integer value, based on a static member variable that is increased every time a new one is created.

Nobody actually *uses* this ID for anything, and as far as I can tell there's not even a 'getInstanceID' function on the class. 
 Perhaps it's sometimes used while debugging?  Anyway, in case it turns out to be useful later, I'll go ahead and fix this (and fix the save/load bit too).

Reply:
Ah OK, that makes sense.

Reply:
OK, updated, fixed, and new tests added.